astroquery/gaiatap/core.py

```diff
@@ -0,0 +1,383 @@
+        ----------
+        jobfilter : JobFilter, optional, default None
+            job filter
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of Job objects
+        """
+        return self.__gaiatap.search_async_jobs(jobfilter, verbose)
+
+    def list_async_jobs(self, verbose=False):
+        """Returns all the asynchronous jobs
+        TAP & TAP+
+        
+        Parameters
+        ----------
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of Job objects
+        """
+        return self.__gaiatap.list_async_jobs(verbose)
+    
+    def query_object(self, coordinate, radius=None, width=None, height=None, async=False, verbose=False):
+        """Launches a job
+        TAP & TAP+
+        
+        Parameters
+        ----------
+        coordinate : astropy.coordinates, mandatory
+            coordinates center point
+        radius : astropy.units, required if no 'width' nor 'height' are provided
+            radius (deg)
+        width : astropy.units, required if no 'radius' is provided
+            box width
+        height : astropy.units, required if no 'radius' is provided
+            box height
+        async : bool, optional, default 'False'
+            executes the query (job) in asynchronous/synchronous mode (default synchronous)
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        The job results (astropy.table).
+        """
+        return self.__gaiatap.query_object(coordinate, radius=radius, width=width, height=height, async=async, verbose=verbose)
+    
+    def query_object_async(self, coordinate, radius=None, width=None, height=None, verbose=False):
+        """Launches a job (async)
+        TAP & TAP+
+        
+        Parameters
+        ----------
+        coordinate : astropy.coordinates, mandatory
+            coordinates center point
+        radius : astropy.units, required if no 'width' nor 'height' are provided
+            radius
+        width : astropy.units, required if no 'radius' is provided
+            box width
+        height : astropy.units, required if no 'radius' is provided
+            box height
+        async : bool, optional, default 'False'
+            executes the query (job) in asynchronous/synchronous mode (default synchronous)
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        The job results (astropy.table).
+        """
+        return self.__gaiatap.query_object_async(coordinate, radius=radius, width=width, height=height, verbose=verbose)
+    
+    def query_region(self, coordinate, radius=None, width=None):
+        print ("Not available")
```

Comment:
For these 'not available' methods, could you just leave them commented out for now?  I wouldn't want users tricked into thinking that method is available when it's not yet.

Also, we prefer to `raise NotImplementedError()` rather than print a statement like this